plugin/core/src/main/groovy/com/novoda/gradle/release/internal/AndroidAttachments.groovy

```diff
@@ -41,13 +41,24 @@ class AndroidAttachments extends MavenPublicationAttachments {
 
     private static Task androidJavadocsJarTask(Project project, String publicationName, def variant) {
         Javadoc javadoc = project.task("javadoc${publicationName.capitalize()}", type: Javadoc) { Javadoc javadoc ->
-            javadoc.source = variant.javaCompiler.source
-            javadoc.classpath = variant.javaCompiler.classpath
+            if (variant.hasProperty('javaCompileProvider')) {
+                // Gradle 4.10.1+ Android 3.3.0+
+                javadoc.source = variant.javaCompilerProvider().get().source
```

Comment:
This usage does not provide the benefit of provider. The provider is there for lazy task configuration and calling get is not recommended during configuration time.

I don't know how feasible it is but the right way would be:

Use `project.tasks.register` instead of `project.task`. Then this method will return `Provider<Task>` instead. That will cascade.

Would be nice to check if it is easy to do.